packages/union-component/src/components/DonationPaymentForm.tsx

```diff
@@ -98,6 +99,14 @@ const DonationPaymentForm: React.FC<Props> = ({
     setFormData((state) => ({ ...state, phoneNumber }));
   };
 
+  const setFundId = useCallback(
+    (fundId: string) => setFormData((state) => ({ ...state, fundId })),
+    []
+  );
+
+  const onChangeFund = (e: React.ChangeEvent<HTMLSelectElement>) =>
```

Comment:
I think you shouldn't use this one rather `onChangeInput` perhaps we can update the function signature and name but essentially `setFormData((state) => ({ ...state, [e.target.name]: e.target.value }));` so you will update the form data accordingly no need of `setFundId` nor `onChangeFund`

Reply:
In fact, as the way it is implemented now, I think that you will be able to submit without setting a `fundId`, it is that what we want? _(either way the comment above apply)_